src/useSharingIntent.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect } from 'react';
+import { AppState, Linking, NativeModules, Platform } from 'react-native';
+
+import { sortData } from './utils';
+const { ReceiveSharingIntent } = NativeModules;
+
+const isIos = Platform.OS === 'ios';
+
+export function useSharingIntent(
+  handler: Function,
+  errorHandler: Function,
+  protocol: string = 'ShareMedia'
+) {
+  const getFileNames = useCallback(
+    (url: string) => {
+      if (isIos) {
+        ReceiveSharingIntent.getFileNames(url)
+          .then((data: any) => {
+            const files = sortData(data);
+            handler(files);
+          })
+          .catch((e: any) => errorHandler(e));
+      } else {
+        ReceiveSharingIntent.getFileNames()
+          .then((fileObject: any) => {
+            const files = Object.keys(fileObject).map((k) => fileObject[k]);
```

Comment:
You can move the content of this and the one in iOS into single function in which you can add a condition isIos

Reply:
I tried adding it like that at first but I got some TS errors, will try again to see if I made some mistakes